src/codegen/read_options.rs

```diff
@@ -94,8 +90,7 @@ fn generate_enum(input: &DeriveInput, tla: &TopLevelAttrs, en: &DataEnum) -> Res
         return Ok(enum_match)
     }
 
-    // return_all_errors is default behavior
-    let return_all_errors = *tla.return_all_errors || !*tla.return_all_errors;
+    let return_all_errors = *tla.return_all_errors;
```

Comment:
While the logic here was broken, this is not how to fix it. return_all_errors should be the default error handling strategy if return_unexpected_error isn't explicitly opted into

Reply:
Thanks! Obviously I was not taking the time to realise `return_unexpected_error` was the opposite flag, even though it’s clear if I’d just looked upwards a few lines 😞. I looked only at the attribute docs and was confused about what was supposed to be happening here since `return_all_errors` had no example and `return_unexpected_error` is not documented. I’m not sure why `return_all_errors` is an attribute at all, since if it is the default, how could it ever be turned off and then on again by an attribute? `return_unexpected_error` seems like the only flag that should exist in this case. Let me know your thoughts about this—maybe there’s something about the design of top-level attributes that I don’t understand—and I’ll fix it properly.

Reply:
Honestly yes the return_all_errors is a useless attribute I just allowed it for the sake of being explicit if the user desired. Obviously return_unexpected_errors never got implemented so it looks pretty dumb, but the idea was whitelisting "expected" errors, such as invalid magic, then when it hits an unexpected error it is treated as unrecoverable and bubbled up.

Imo any fix you go with will be fine so long as return_all_errors defaults to true. The "correct" fix is just like. `return_all_errors || !return_unexpected_error` but obviously that's not really different from just `= !return_unexpected_error` so it doesn't realllllly matter.

Reply:
I’ve updated this line so it’s just `!*tla.return_unexpected_error` now. Referencing an attribute which doesn’t do anything… seems to me like it’d make it seem like it… does something, when it doesn’t. If it were up to me I’d just rip it out entirely now that I know it’s actually not doing anything and the alternate error handling code is not complete, but I don’t want to scope creep this PR. Let me know if there’s anything else you would like to change here.